deriva/seo/sitemap_builder.py

```diff
@@ -0,0 +1,330 @@
+        for fk_row in fk_rows:
+            entry = rdict.get(fk_row["RID"])
+            if entry is not None:
+                fk_time = self.ermrest_time_to_float(fk_row["RMT"])
+                row = entry.get("row")
+                entry_time = entry.get("rmt_float")
+                if entry_time is None:
+                    entry_time = self.ermrest_time_to_float(row["RMT"])
+                if fk_time > entry_time:
+                    row["RMT"] = fk_row["RMT"]
+                    entry["rmt_float"] = fk_time
+                else:
+                    entry["rmt_float"] = entry_time
+
+    @staticmethod
+    def ermrest_time_to_float(ermrest_time):
+        """Converts a time string as returned by ermrest to a floating-point number
+        """
+        return(time.mktime(time.strptime(ermrest_time, "%Y-%m-%dT%H:%M:%S.%f%z")))
+
+    def get_table(self, sname, tname):
+        """Gets a datapath table
+        """
+        return self.pb.schemas.get(sname).tables.get(tname)
+
+    def get_column(self, table, column_name):
+        """Gets a datapath column
+        """
+        return table.column_definitions.get(column_name)
+
+    def set_table_spec_data(self, spec, rows):
+        """Add data to a table spec. This should be used if you want to associate
+           image data with the rows of this table, or if you want to include
+           only a subset of the table's rows in the sitemap. Otherwise, you can
+           just use populate=True in your call to add_table_spec().
+
+        :param spec: a table spec returned by add_table_spec()
+        :param rows: an array of dictionary objects corresponding to the rows
+           of the table. Each row MUST have RID and RMT elements and MAY have
+           an "images" element. If "images" is present, each entry should have:
+              "image_url" : the absolute URL of the image (jpeg, etc.)
+              "image_caption" : a caption for the image (optional)
+              "image_title" : a title for the image (optional)
+
+           for example::
+
+              [
+                 {
+                    "RID": '1-2345',
+                    "RMT": '018-12-03T18:29:38.348231-08:00'},
+                 {
+                    "RID": '1-2346',
+                    "RMT": '018-12-03T18:29:38.348231-08:00',
+                    "images": [
+                        {
+                           "image_url": "https://myhost.org/hatrac/images/image1.jpg",
+                           "image_title": "MyProject Image 2-3456: An Awesome Image",
+                           "image_caption": "This is an awesome image"
+                        },
+                        {
+                           "image_url": "https://myhost.org/hatrac/images/image2.jpg",
+                        }
+                    ]
+                 }
+              ]
+        """
+        spec["data"] = rows
+
+    def find_incoming_fkeys(self, target_table):
```

Comment:
I'm a little unclear about the use for this? Is this different than the `Table.referenced_by` property in `ermrest_model`?

Reply:
I didn't know that property existed! That's convenient.